Makefile

```diff
@@ -165,146 +165,81 @@ protoc:
 # mockgen must be installed on the system to make this work.
 # Install it by running:
 # `go get github.com/golang/mock/mockgen`.
-mockgen = $(GOPATH)/bin/mockgen
+mockgen = GO111MODULES=on $(GOPATH)/bin/mockgen
+
+define kraken_mockgen
+	mkdir -p mocks/$(1)
+	$(mockgen) \
+		-destination=mocks/$(1)/$(shell tr '[:upper:]' '[:lower:]' <<< $(3)).go \
+		-package $(2) \
+		$(PROJECT_ROOT)/$(1) $(3)
+
+endef
 
 .PHONY: mocks
 mocks:
 	rm -rf mocks
 	mkdir -p $(GOPATH)/bin
 
-	mkdir -p mocks/lib/hashring
-	$(mockgen) \
-		-destination=mocks/lib/hashring/mocks.go \
-		-package mockhashring \
-		github.com/uber/kraken/lib/hashring Ring,Watcher
-
-	mkdir -p mocks/lib/backend/s3backend
-	$(mockgen) \
-		-destination=mocks/lib/backend/s3backend/mocks.go \
-		-package mocks3backend \
-		github.com/uber/kraken/lib/backend/s3backend S3
-
+	$(call kraken_mockgen,lib/backend/s3backend,mocks3backend,S3)
```

Comment:
Nit: the name `kraken_mockgen` feels sort of clunky, what about `add_mock`?

Reply:
Done.